### ISSUE TITLE
Typed array constructors should accept generic iterables

### ISSUE THREAD

**Issue by abchatra** [Microsoft/ChakraCore#22]
Following test should should return true.
```javascript
var arr = [1,2,3];
var iterableObj = {
    [Symbol.iterator]: ()=> {
        return {
            next: () => {
                return {
                    value: arr.shift(),
                    done: arr.length == 0
                };
            }
        };
    }
};
var arr = new Uint8Array(iterableObj);
print(arr);
```
Expected: 1,2
Actual: TypeError

See duplicate OS bug [5027123](https://microsoft.visualstudio.com/web/wi.aspx?pcguid=cb55739e-4afe-46a3-970f-1b49d8ee7564&id=5027123) 

I believe following [kangax](https://kangax.github.io/compat-table/es6/#test-typed_arrays_constructors_accept_generic_iterables) test case is dupe of this too:
```javascript
(function(){
var constructors = [
  'Int8Array',
  'Uint8Array',
  'Uint8ClampedArray',
  'Int16Array',
  'Uint16Array',
  'Int32Array',
  'Uint32Array',
  'Float32Array',
  'Float64Array'
];
for(var i = 0; i < constructors.length; i++){
  var arr = new global[constructors[i]](__createIterableObject([1, 2, 3]));
  if(arr.length !== 3 || arr[0] !== 1 || arr[1] !== 2 || arr[2] !== 3)return false;
}
return true;
})();
````

**Comment by suwc** [comment]
https://github.com/Microsoft/ChakraCore/pull/233

**Comment by obastemur** [comment]
Guess this needs to be closed.

**Comment by ianwjhalliday** [comment]
Fixed by #233
